static/app/components/events/highlights/highlightsDataSection.tsx

```diff
@@ -1,20 +1,217 @@
+}: HighlightsSectionProps) {
   const hasNewTagsUI = useHasNewTagsUI();
+  const organization = useOrganization();
+  const containerRef = useRef<HTMLDivElement>(null);
+  const columnCount = useIssueDetailsColumnCount(containerRef);
+  const {
+    isLoading,
+    data: detailedProject,
+    refetch,
+  } = useDetailedProject({
+    orgSlug: organization.slug,
+    projectSlug: project.slug,
+  });
+
   if (!hasNewTagsUI) {
     return null;
   }
-  // TODO(Leander): When a design is confirmed, remove this usage of ContextSummary
-  return <ContextSummary event={event} />;
+
+  const highlightContext = detailedProject?.highlightContext ?? {};
+  const highlightTags = detailedProject?.highlightTags ?? [];
+
+  // The API will return default values for tags/context. The only way to have none is to set it to
+  // empty yourself, meaning the user has disabled highlights
+  const hasDisabledHighlights =
+    Object.values(highlightContext).flat().length === 0 && highlightTags.length === 0;
+
+  const viewAllButton = viewAllRef ? (
+    <Button
+      onClick={() => viewAllRef?.current?.scrollIntoView({behavior: 'smooth'})}
+      size="xs"
+    >
+      {t('View All')}
+    </Button>
+  ) : null;
+
+  const highlightContextDataItems = getHighlightContextData({
+    event,
+    project,
+    organization,
+    highlightContext,
+  });
+  const highlightContextRows = highlightContextDataItems.reduce<React.ReactNode[]>(
+    (rowList, {alias, data}, i) => {
+      const meta = getContextMeta(event, alias);
+      const newRows = data.map((item, j) => (
+        <HighlightContextContent
+          key={`highlight-ctx-${i}-${j}`}
+          meta={meta}
+          item={item}
+          alias={alias}
+          config={{includeAliasInSubject: true}}
+        />
+      ));
+      return [...rowList, ...newRows];
+    },
+    []
+  );
+
+  const highlightTagItems = getHighlightTagData({event, highlightTags});
+  const highlightTagRows = highlightTagItems.map((content, i) => (
+    <EventTagsTreeRow
+      key={`highlight-tag-${i}`}
+      content={content}
+      event={event}
+      tagKey={content.originalTag.key}
+      projectSlug={project.slug}
+    />
+  ));
+
+  const rows = [...highlightTagRows, ...highlightContextRows];
+  const columns: React.ReactNode[] = [];
+  const columnSize = Math.ceil(rows.length / columnCount);
+  for (let i = 0; i < rows.length; i += columnSize) {
+    columns.push(
+      <HighlightColumn key={`highlight-column-${i}`}>
+        {rows.slice(i, i + columnSize)}
+      </HighlightColumn>
+    );
+  }
+
+  function openEditHighlightsModal() {
+    openModal(
+      deps => (
+        <EditHighlightsModal
+          event={event}
+          highlightContext={highlightContext}
+          highlightTags={highlightTags}
+          project={detailedProject ?? project}
+          highlightPreset={detailedProject?.highlightPreset}
+          {...deps}
+        />
+      ),
+      {modalCss: highlightModalCss, onClose: refetch}
```

Comment:
Instead of `refetch` on close, I'd suggest using `setApiQueryData` in the `useMutation` `onSuccess`. That way the change will be reflected immediately. Usage is documented here: https://develop.sentry.dev/frontend/network-requests/#updating-your-query-data